server/src/main/java/org/apache/druid/server/coordinator/CoordinatorDynamicConfig.java

```diff
@@ -231,32 +231,35 @@ public int getMaxSegmentsInNodeLoadingQueue()
   }
 
   /**
-   * Historical nodes list in maintenance mode. Coordinator doesn't assign new segments on those nodes and moves
-   * segments from those nodes according to a specified priority.
+   * List of historical servers to 'decommission'. Coordinator will not assign new segments to 'decomissioning' servers,
+   * and segments will be moved away from them to be placed on 'active' servers at the maximum rate specified by
+   * {@link CoordinatorDynamicConfig#getDecommissioningMaxPercentOfMaxSegmentsToMove}.
    *
    * @return list of host:port entries
    */
   @JsonProperty
-  public Set<String> getHistoricalNodesInMaintenance()
+  public Set<String> getDecommissioningNodes()
   {
-    return historicalNodesInMaintenance;
+    return decommissioningNodes;
   }
 
   /**
-   * Priority of segments from servers in maintenance. Coordinator takes ceil(maxSegmentsToMove * (priority / 10))
-   * from servers in maitenance during balancing phase, i.e.:
-   * 0 - no segments from servers in maintenance will be processed during balancing
-   * 5 - 50% segments from servers in maintenance
-   * 10 - 100% segments from servers in maintenance
-   * By leveraging the priority an operator can prevent general nodes from overload or decrease maitenance time
-   * instead.
+   * The maximum number of segments that may be moved away from 'decommissioning' servers to non-decommissioning
+   * (that is, active) servers during one Coordinator's run. This value is relative to the total maximum segment
+   * movements allowed during one run which is determined by `{@link CoordinatorDynamicConfig#getMaxSegmentsToMove()}.
    *
-   * @return number in range [0, 10]
+   * If `decommissioningMaxPercentOfMaxSegmentsToMove` is 0, segments will neither be moved from _or to_ 'decommissioning'
+   * servers, effectively putting them in a sort of 'maintenance' mode that will not participate in balancing or
+   * assignment by load rules. Decommissioning can also become stalled if there are no available active servers to place
+   * the segments. By leveraging decommissioning percent, an operator can prevent active servers from overload by
```

Comment:
Suggested "By leveraging the maximum percent of decommissioning segment movements, ..."